### PR TITLE
refactor: response time class api

### PR DESCRIPTION
**Pull request by rokamu623**
## Description
Implement `to_best_case_records()` and `to_worst_case_records()` in the `ResponseTime class`, which are APIs in the same format as `to_records()` in `Latency` class.

## Related links
Jira: https://tier4.atlassian.net/browse/RT2-886

## Pre-review checklist for the PR author

- [x] I've confirmed the [contribution guidelines](https://github.com/tier4/caret/blob/main/.github/CONTRIBUTING.md).
- [x] The PR is ready for review.

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR has been properly tested.
- [ ] The PR has been reviewed.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] (Optional) The PR has been properly tested with CARET_report verification.
- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

### CODE REVIEW

**Comment by rokamu623** [comment]
@taro-yu 
Please review.
The pytest error will be resolved by #311 or #312 .

File: src/caret_analyze/record/records_service/response_time.py [hunk 1]
```diff
@@ -759,6 +745,42 @@ def _to_timeseries(self, input_column, output_column):
...
+        input_column = records.columns[1]
+        output_column = records.columns[-1]
+        return self._to_records(input_column, output_column)
+
+    def to_worst_case_records(self) -> RecordsInterface:
+        records = self._records.to_range_records()
+        input_column = records.columns[0]
+        output_column = records.columns[-1]
+        return self._to_records(input_column, output_column)
+
```
Comment (taro-yu) [inline src/caret_analyze/record/records_service/response_time.py:759]: @rokamu623 
I think that using` to_range_records` in best_case and worst_case are not best way just for column name.
Although there is no way to get column names in ResponseTime class, it is better that, for example, create a new variable `self._input_records = records` in the ResponseTime class and then write as `input_column = f'{self._input_records.columns[0]}_min or max'.`
However, it is a little more complicated to write than before.

File: src/caret_analyze/record/records_service/response_time.py [hunk 2]
```diff
@@ -759,6 +745,42 @@ def _to_timeseries(self, input_column, output_column):
...
+    def _to_records(self, input_column, output_column) -> RecordsInterface:
+        records: RecordsInterface = self._create_empty_records(input_column)
+
+        range_records = self._records.to_range_records()
+        t_in = range_records.get_column_series(input_column)
+        t_out = range_records.get_column_series(output_column)
```
Comment (taro-yu) [inline src/caret_analyze/record/records_service/response_time.py:765]: It's better to change t_in/out to timestamps_in/out or somethings more comprehensible.

File: src/caret_analyze/record/records_service/response_time.py [hunk 3]
```diff
@@ -759,6 +745,42 @@ def _to_timeseries(self, input_column, output_column):
...
+        records = self._records.to_range_records()
+        input_column = records.columns[1]
+        output_column = records.columns[-1]
+        return self._to_records(input_column, output_column)
+
+    def to_worst_case_records(self) -> RecordsInterface:
+        records = self._records.to_range_records()
+        input_column = records.columns[0]
+        output_column = records.columns[-1]
+        return self._to_records(input_column, output_column)
```
Comment (taro-yu) [inline src/caret_analyze/record/records_service/response_time.py:758]: Although it is also used in the function from line 743, I think that using to_range_records is not best way just for column name.
Although there is no way to get column names in ResponseTime class, it is better that, for example, create a new variable self._input_records = records in the ResponseTime class and then write as input_column = f'{self._input_records.columns[0]}_min or max'.
However, it is a little more complicated to write than before.

File: src/test/record/records_service/test_response_time.py [hunk 4]
```diff
@@ -220,26 +151,10 @@ def test_double_flow_cross_case(self):
...
-        result = to_dict(response.to_response_records())
-        assert result == expect_raw
-
-        expect_raw = [
-            {'start': 0, 'end': 3},
-            {'start': 0, 'end': 5},
-            {'start': 2, 'end': 3},
-        ]
-
-        result = to_dict(response.to_records(all_pattern=True))
```
Comment (taro-yu) [inline src/test/record/records_service/test_response_time.py:242]: In this test case, it seems not to consider the case: 'end' : 5 in line 238.

File: src/test/record/records_service/test_response_time.py [hunk 5]
```diff
@@ -220,26 +151,10 @@ def test_double_flow_cross_case(self):
...
+        result = to_dict(response.to_stacked_bar())
         assert result == expect_raw
```
Comment (taro-yu) [inline src/test/record/records_service/test_response_time.py:158]: In this test case, it seems not to consider the case: 'end' : 5 in line 238.

File: src/test/record/records_service/test_response_time.py [hunk 6]
```diff
@@ -134,39 +102,20 @@ def test_cross_flow_case(self):
...
-            {'start': 0, 'end': 6},
-            {'start': 0, 'end': 8},
-            {'start': 0, 'end': 10},
             {'start': 3, 'end': 4},
-            {'start': 4, 'end': 8},
             {'start': 6, 'end': 6},
         ]
-        result = to_dict(response.to_records(all_pattern=True))
+        result = to_dict(response.to_best_case_stacked_bar())
         assert result == expect_raw
```
Comment (taro-yu) [inline src/test/record/records_service/test_response_time.py:117]: Is it intentional that 'start' : 4 is not taken into account?

File: src/test/record/records_service/test_response_time.py (hunk 5 above)
Comment (taro-yu) [inline src/test/record/records_service/test_response_time.py:158]: In this test case, it seems not to consider the case: 'end' : 5 in line 238.
Is this OK?

File: src/test/record/records_service/test_response_time.py (hunk 6 above)
Comment (rokamu623) [reply src/test/record/records_service/test_response_time.py:117]: The `cross_case` may be different from the desired specification. (The implement may wrong.)

However, as this PR is a refactoring, this investigation and correction will not be conducted.

File: src/test/record/records_service/test_response_time.py (hunk 5 above)
Comment (rokamu623) [reply src/test/record/records_service/test_response_time.py:158]: The `cross_case` may be different from the desired specification. (The implement may wrong.)

However, as this PR is a refactoring, this investigation and correction will not be conducted.

File: src/caret_analyze/record/records_service/response_time.py (hunk 2 above)
Comment (rokamu623) [reply src/caret_analyze/record/records_service/response_time.py:764]: This PR will not be changed in this PR as it is written in accordance with existing code.

File: src/caret_analyze/record/records_service/response_time.py (hunk 3 above)
Comment (rokamu623) [reply src/caret_analyze/record/records_service/response_time.py:757]: This PR will not be changed in this PR as it is written in accordance with existing code.

**Comment by rokamu623** [comment]
@nabetetsu 
Please check this PR.

**Review by taro-yu — DISMISSED** [review]
LGTM!

**Review by nabetetsu — APPROVED** [review]
[rt2-886_result.zip](https://github.com/tier4/CARET_analyze/files/12462544/rt2-886_result.zip)
LGTM :+1:
I have confirmed that there is no difference between the visualization results of Stacked bar and Timeseries before and after this PR's modification.

**Pull request merged**